### PR TITLE
Fix usNIC CIDs

### PR DESCRIPTION
**Pull request by jsquyres**
See individual commit messages for details.

### CODE REVIEW

**Comment by jsquyres** [comment]
@j-xiong How does one get feedback on the ci/cloudbees/stage/Summary and continuous-integration/intel/jenkins CI failures?  Clicking the Details links seem to try to go to internal Intel URLs.

**Comment by j-xiong** [comment]
@jsquyres Unfortunately only someone from Intel can see the details.

For this PR, I checked the error yesterday and saw infrastructure related job cancellations and restarted the run. The new failure is a single transient mpichtestsuite failure with the tcp provider, which is a known issue and is unrelated to the PR.

**Review by j-xiong — APPROVED** [review]

File: prov/usnic/src/usdf_pep.c [hunk 1]
```diff
@@ -478,17 +478,17 @@ usdf_pep_grow_backlog(struct usdf_pep *pep)
 
 	extra = sizeof(struct usdf_connreq_msg) + pep->pep_cr_max_data;
 
+	pthread_spin_lock(&pep->pep_cr_lock);
 	while (pep->pep_cr_alloced < pep->pep_backlog) {
 		crp = calloc(1, sizeof(*crp) + extra);
 		if (crp == NULL) {
 			return -FI_ENOMEM;
```
Comment (shefty) [inline prov/usnic/src/usdf_pep.c:485]: This returns while holding the spinlock

Comment (j-xiong) [reply prov/usnic/src/usdf_pep.c:485]: ahh, missed this one. Thanks!

Comment (jsquyres) [reply prov/usnic/src/usdf_pep.c:485]: Faceplam.  I'll fix; thanks!

Comment (jsquyres) [reply prov/usnic/src/usdf_pep.c:486]: Fixed.

**Comment by jsquyres** [comment]
Forgive me for being out of the libfabric loop for so long -- what's the merge policy these days?

**Comment by j-xiong** [comment]
@jsquyres Generally we expect CI runs to pass and receive at least one approval before merge. However sometimes exceptions are made (e.g. totally unrelated CI failure).

**Pull request merged**